apps/user-cdc/src/main/java/it/pagopa/selfcare/user/event/UserInstitutionCdcService.java

```diff
@@ -68,11 +86,31 @@ private ReactiveMongoCollection<UserInstitution> getCollection() {
     }
 
     protected void consumerUserInstitutionRepositoryEvent(ChangeStreamDocument<UserInstitution> document) {
+
         assert document.getFullDocument() != null;
+        assert document.getDocumentKey() != null;
+
         userInstitutionRepository.updateUser(document.getFullDocument())
+                .onFailure().retry().withBackOff(Duration.ofSeconds(retryMinBackOff), Duration.ofHours(retryMaxBackOff)).atMost(maxRetry)
                 .subscribe().with(
-                        result -> log.info("UserInfo collection successfully updated"),
-                        failure -> log.error("Error during UserInfo collection updating, message:" + failure.getMessage()));
+                        result -> {
+                            constructMapAndTrackEvent(document.getDocumentKey().toJson(), "TRUE", USERINSTITUTION_SUCCESS_MECTRICS, USERINSTITUTION_COUNT_MECTRICS);
```

Comment:
you could avoid COUNT_METRICS because we could count success or failure summing USERINSTITUTION_SUCCESS_MECTRICS or USERINSTITUTION_FAILURE_MECTRICS